completion.go

```diff
@@ -65,11 +65,22 @@ func (c *CompletionManager) GetSuggestions() []Suggest {
 	return c.tmp
 }
 
+// GetSuggestions returns the list of suggestion.
+func (c *CompletionManager) GetSelectedIdx() int {
+	c.mu.RLock()
```

Comment:
```
c.mu.RLock()
	defer c.mu.RUnlock()
```

Reply:
I think this spot here is fine, I'm acquiring a read lock and releasing it at the end.

But at other places you're right, I just locked everything with a write lock as it became too difficult to read to separate all of the read vs write operations. And you also have to be careful not to lock twice by calling another method that also does locking, or we could end up in a deadlock.